Shell/Formatter.cpp

```diff
@@ -0,0 +1,648 @@
+    current_builder().append('(');
+    TemporaryChange<const AST::Node*> parent { m_parent_node, node };
+
+    auto first = true;
+    for (auto& arg : node->arguments()) {
+        if (!first)
+            current_builder().append(' ');
+        first = false;
+        current_builder().append(arg.name);
+    }
+
+    current_builder().append(") {");
+
+    {
+        TemporaryChange indent { m_current_indent, m_current_indent + 1 };
+        insert_separator();
+
+        if (node->block())
+            node->block()->visit(*this);
+    }
+
+    insert_separator();
+    current_builder().append('}');
+}
+
+void Formatter::visit(const AST::ForLoop* node)
+{
+    test_and_update_output_cursor(node);
+    current_builder().append("for ");
+    TemporaryChange<const AST::Node*> parent { m_parent_node, node };
+
+    if (node->variable_name() != "it") {
+        current_builder().append(node->variable_name());
+        current_builder().append(" in ");
+    }
+
+    if (node->iterated_expression())
+        node->iterated_expression()->visit(*this);
+
+    current_builder().append(" {");
+    {
+        TemporaryChange indent { m_current_indent, m_current_indent + 1 };
+
+        insert_separator();
+
+        if (node->block())
+            node->block()->visit(*this);
+    }
+
+    insert_separator();
+    current_builder().append('}');
+}
+
+void Formatter::visit(const AST::Glob* node)
+{
+    test_and_update_output_cursor(node);
+    current_builder().append(node->text());
+}
+
+void Formatter::visit(const AST::Execute* node)
+{
+    test_and_update_output_cursor(node);
+    auto& builder = current_builder();
+    TemporaryChange<const AST::Node*> parent { m_parent_node, node };
+
+    auto old_options = m_options;
+
+    m_options.explicit_parentheses = false;
+
+    if (node->does_capture_stdout()) {
+        builder.append("$");
+        m_options.explicit_parentheses = true;
+    }
+
+    NodeVisitor::visit(node);
+
+    builder.append(m_temp_builders.last().string_view());
+
+    m_options = move(old_options);
+}
+
+void Formatter::visit(const AST::IfCond* node)
+{
+    test_and_update_output_cursor(node);
+
+    current_builder().append("if ");
+    TemporaryChange<const AST::Node*> parent { m_parent_node, node };
+
+    if (node->condition())
+        node->condition()->visit(*this);
+
+    current_builder().append(" {");
+    {
+        TemporaryChange indent { m_current_indent, m_current_indent + 1 };
+
+        insert_separator();
+        if (node->true_branch())
+            node->true_branch()->visit(*this);
+    }
+
+    insert_separator();
+    current_builder().append('}');
+
+    if (node->false_branch()) {
+        if (node->false_branch()->class_name() != "IfCond") {
+            current_builder().append(" else {");
+            {
+                TemporaryChange indent { m_current_indent, m_current_indent + 1 };
+
+                insert_separator();
+                node->false_branch()->visit(*this);
+            }
```

Comment:
Since it's been repeated multiple times, I believe adding the block was intentional. I'd personally not introduce unnecessary blocks if I don't have to but feel free to ignore this comment if you like it that way.